libptable/ptable.c

```diff
@@ -127,13 +116,18 @@ static int ptable_partVerify(const ptable_t *ptable, const ptable_part_t *part,
 static int ptable_verify(const ptable_t *ptable, uint32_t memsz, uint32_t blksz)
 {
 	uint32_t size, i;
+	int crcCheck = 1;
 
-#if PTABLE_CHECKSUM
-	/* Verify header checksum */
-	if (ptable->crc != ptable_crc32(ptable, offsetof(ptable_t, crc))) {
-		return -1;
+	if (ptable->version == 0 || ptable->version == 0xff) {
+		/* Disable CRC check for legacy ptables */
+		crcCheck = 0;
+	}
+	else {
```

Comment:
maybe write `if (crcCheck)` directly (there might be some additional cases when crc check should be disabled in the future)

Reply:
Done